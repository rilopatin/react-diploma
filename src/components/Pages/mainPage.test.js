import React from 'react'
import MainPage from './mainPage'
import {shallow} from 'enzyme'


describe('Testing <MainPage/>', () => {
    const main = shallow(<MainPage/>)
    it('MainPage has been rendered correctly', () => {
        expect(main).toMatchSnapshot()
    });

    it('CharacterPage state "error" is false', () => {
        expect(main.state.error).toBeFalsy()
    });
})