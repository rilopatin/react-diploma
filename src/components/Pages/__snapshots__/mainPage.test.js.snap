// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <MainPage/> MainPage has been rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="preview"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-lg-6"
              >
                <Header />
              </div>
            </div>
            <div
              className="row"
            >
              <JumbotronComponent />
            </div>
          </div>
        </div>,
        <AboutUsComponent />,
        <div
          className="best"
        >
          <div
            className="container"
          >
            <div
              className="title"
            >
              Our best
            </div>
            <div
              className="row"
            >
              <div
                className="col-lg-10 offset-lg-1"
              >
                <ItemListComponent
                  classes={
                    Array [
                      "best__item",
                      "best__item-title",
                      "best__item-price",
                      "best__wrapper",
                    ]
                  }
                  details={true}
                  getData={[Function]}
                  renderItem={[Function]}
                />
              </div>
            </div>
          </div>
        </div>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-lg-6"
              >
                <Header />
              </div>
            </div>
            <div
              className="row"
            >
              <JumbotronComponent />
            </div>
          </div>,
          "className": "preview",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="row"
              >
                <div
                  className="col-lg-6"
                >
                  <Header />
                </div>
              </div>,
              <div
                className="row"
              >
                <JumbotronComponent />
              </div>,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="col-lg-6"
                >
                  <Header />
                </div>,
                "className": "row",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Header />,
                  "className": "col-lg-6",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <JumbotronComponent />,
                "className": "row",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <div
              className="title"
            >
              Our best
            </div>
            <div
              className="row"
            >
              <div
                className="col-lg-10 offset-lg-1"
              >
                <ItemListComponent
                  classes={
                    Array [
                      "best__item",
                      "best__item-title",
                      "best__item-price",
                      "best__wrapper",
                    ]
                  }
                  details={true}
                  getData={[Function]}
                  renderItem={[Function]}
                />
              </div>
            </div>
          </div>,
          "className": "best",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
              >
                Our best
              </div>,
              <div
                className="row"
              >
                <div
                  className="col-lg-10 offset-lg-1"
                >
                  <ItemListComponent
                    classes={
                      Array [
                        "best__item",
                        "best__item-title",
                        "best__item-price",
                        "best__wrapper",
                      ]
                    }
                    details={true}
                    getData={[Function]}
                    renderItem={[Function]}
                  />
                </div>
              </div>,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Our best",
                "className": "title",
              },
              "ref": null,
              "rendered": "Our best",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="col-lg-10 offset-lg-1"
                >
                  <ItemListComponent
                    classes={
                      Array [
                        "best__item",
                        "best__item-title",
                        "best__item-price",
                        "best__wrapper",
                      ]
                    }
                    details={true}
                    getData={[Function]}
                    renderItem={[Function]}
                  />
                </div>,
                "className": "row",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    null,
                    <ItemListComponent
                      classes={
                        Array [
                          "best__item",
                          "best__item-title",
                          "best__item-price",
                          "best__wrapper",
                        ]
                      }
                      details={true}
                      getData={[Function]}
                      renderItem={[Function]}
                    />,
                  ],
                  "className": "col-lg-10 offset-lg-1",
                },
                "ref": null,
                "rendered": Array [
                  null,
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "classes": Array [
                        "best__item",
                        "best__item-title",
                        "best__item-price",
                        "best__wrapper",
                      ],
                      "details": true,
                      "getData": [Function],
                      "renderItem": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="preview"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col-lg-6"
                >
                  <Header />
                </div>
              </div>
              <div
                className="row"
              >
                <JumbotronComponent />
              </div>
            </div>
          </div>,
          <AboutUsComponent />,
          <div
            className="best"
          >
            <div
              className="container"
            >
              <div
                className="title"
              >
                Our best
              </div>
              <div
                className="row"
              >
                <div
                  className="col-lg-10 offset-lg-1"
                >
                  <ItemListComponent
                    classes={
                      Array [
                        "best__item",
                        "best__item-title",
                        "best__item-price",
                        "best__wrapper",
                      ]
                    }
                    details={true}
                    getData={[Function]}
                    renderItem={[Function]}
                  />
                </div>
              </div>
            </div>
          </div>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col-lg-6"
                >
                  <Header />
                </div>
              </div>
              <div
                className="row"
              >
                <JumbotronComponent />
              </div>
            </div>,
            "className": "preview",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="row"
                >
                  <div
                    className="col-lg-6"
                  >
                    <Header />
                  </div>
                </div>,
                <div
                  className="row"
                >
                  <JumbotronComponent />
                </div>,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="col-lg-6"
                  >
                    <Header />
                  </div>,
                  "className": "row",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Header />,
                    "className": "col-lg-6",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <JumbotronComponent />,
                  "className": "row",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
            >
              <div
                className="title"
              >
                Our best
              </div>
              <div
                className="row"
              >
                <div
                  className="col-lg-10 offset-lg-1"
                >
                  <ItemListComponent
                    classes={
                      Array [
                        "best__item",
                        "best__item-title",
                        "best__item-price",
                        "best__wrapper",
                      ]
                    }
                    details={true}
                    getData={[Function]}
                    renderItem={[Function]}
                  />
                </div>
              </div>
            </div>,
            "className": "best",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="title"
                >
                  Our best
                </div>,
                <div
                  className="row"
                >
                  <div
                    className="col-lg-10 offset-lg-1"
                  >
                    <ItemListComponent
                      classes={
                        Array [
                          "best__item",
                          "best__item-title",
                          "best__item-price",
                          "best__wrapper",
                        ]
                      }
                      details={true}
                      getData={[Function]}
                      renderItem={[Function]}
                    />
                  </div>
                </div>,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Our best",
                  "className": "title",
                },
                "ref": null,
                "rendered": "Our best",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="col-lg-10 offset-lg-1"
                  >
                    <ItemListComponent
                      classes={
                        Array [
                          "best__item",
                          "best__item-title",
                          "best__item-price",
                          "best__wrapper",
                        ]
                      }
                      details={true}
                      getData={[Function]}
                      renderItem={[Function]}
                    />
                  </div>,
                  "className": "row",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      null,
                      <ItemListComponent
                        classes={
                          Array [
                            "best__item",
                            "best__item-title",
                            "best__item-price",
                            "best__wrapper",
                          ]
                        }
                        details={true}
                        getData={[Function]}
                        renderItem={[Function]}
                      />,
                    ],
                    "className": "col-lg-10 offset-lg-1",
                  },
                  "ref": null,
                  "rendered": Array [
                    null,
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "classes": Array [
                          "best__item",
                          "best__item-title",
                          "best__item-price",
                          "best__wrapper",
                        ],
                        "details": true,
                        "getData": [Function],
                        "renderItem": [Function],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
